src/pe/tls.rs

```diff
@@ -279,11 +279,11 @@ impl<'a> TlsData<'a> {
                     )));
                 }
                 // Each callback is an VA so convert it to RVA
-                // For x86 compatibility, `usize` is 32-bit, `u64` is 64-bit.
-                // Therefore upcast to u64 first, then downcast the whole var to u32.
-                let callback_rva = (callback - image_base as u64) as usize;
+                let callback_rva = callback - image_base;
                 // Check if the callback is in the image
-                if utils::find_offset(callback_rva, sections, file_alignment, opts).is_none() {
+                if utils::find_offset(callback_rva as usize, sections, file_alignment, opts)
```

Comment:
isn't there technically still an issue on 32-bit systems if `callback_rva is > u32`, and then it gets casted to a `usize` (hence truncating a potentially 64 bit number to 32 bits (sizeof pointer))?

tbh i'm not sure we ever supported analyzing 64-bit binaries on on 32-bit systems, simply due to rust `&[u8]` requiring usize to index, which means if that slice is larger than 32-bits, you cannot access the upper bits in rust.

Reply:
Oh yes, where the case the image must be broken. We shouldn't assume any of TLS callback addresses to be a valid VA/RVA. I'll put the check there.